s3.py

```diff
@@ -220,12 +220,16 @@ def cover_image_url(self, data_source, identifier, filename,
         parts = [identifier.type, identifier.identifier, filename]
         return root + self.key_join(parts)
 
-    def marc_file_url(self, library, lane):
+    def marc_file_url(self, library, lane, start_time=None, end_time=None):
```

Comment:
Your latest changes are great but now either end_time needs to be required, or we need a strategy for when it's `None`, or we need to be okay with `/bucket/None/Lane.mrc`.